packages/compiler/core/library.ts

```diff
@@ -1,6 +1,13 @@
 import { createDiagnosticCreator } from "./diagnostics.js";
 import { CadlLibrary, CadlLibraryDef, CallableMessage, DiagnosticMessages } from "./types.js";
 
+(global as any)._CADL_LIBRARY_LOADED_ = new Set<string>();
+
+/**
+ * @internal List of urls that used `createCadlLibary`. Used to keep track of the loaded version of library and make sure they are compatible.
+ */
+export const LIBRARIES_URL_LOADED = (global as any)._CADL_LIBRARY_LOADED_;
```

Comment:
consider using a `symbol` as key into global object. Like https://github.com/Azure/azure-sdk-for-js/blob/47e53cc9f091bc33ccfc8f443b43fc40cb80a746/sdk/core/core-tracing/src/utils/cache.ts#L11